70_FLIPFluids/gpuRadixSort.h

```diff
@@ -0,0 +1,470 @@
+				};
+				m_device->updateDescriptorSets(std::span(writes, 3), {});
+			}
+			{	// sort: iteration 0 and 2
+				IGPUDescriptorSet::SDescriptorInfo infos[5];
+				infos[0].desc = smart_refctd_ptr(paramsBuffer);
+				infos[0].info.buffer = {.offset = 0, .size = paramsBuffer->getSize()};
+				infos[1].desc = smart_refctd_ptr(dataBuffer);
+				infos[1].info.buffer = {.offset = 0, .size = dataBuffer->getSize()};
+				infos[2].desc = smart_refctd_ptr(tempDataBuffer);
+				infos[2].info.buffer = {.offset = 0, .size = tempDataBuffer->getSize()};
+				infos[3].desc = smart_refctd_ptr(globalHistogramsBuffer);
+				infos[3].info.buffer = {.offset = 0, .size = globalHistogramsBuffer->getSize()};
+				infos[4].desc = smart_refctd_ptr(partitionHistogramBuffer);
+				infos[4].info.buffer = {.offset = 0, .size = partitionHistogramBuffer->getSize()};
+				IGPUDescriptorSet::SWriteDescriptorSet writes[5] = {
+					{.dstSet = m_radixSortDs[0].get(), .binding = 0, .arrayElement = 0, .count = 1, .info = &infos[0]},
+					{.dstSet = m_radixSortDs[0].get(), .binding = 1, .arrayElement = 0, .count = 1, .info = &infos[1]},
+					{.dstSet = m_radixSortDs[0].get(), .binding = 2, .arrayElement = 0, .count = 1, .info = &infos[2]},
+					{.dstSet = m_radixSortDs[0].get(), .binding = 3, .arrayElement = 0, .count = 1, .info = &infos[3]},
+					{.dstSet = m_radixSortDs[0].get(), .binding = 4, .arrayElement = 0, .count = 1, .info = &infos[4]}
+				};
+				m_device->updateDescriptorSets(std::span(writes, 5), {});
+			}
+			{	// sort: iteration 1 and 3
+				IGPUDescriptorSet::SDescriptorInfo infos[5];
+				infos[0].desc = smart_refctd_ptr(paramsBuffer);
+				infos[0].info.buffer = {.offset = 0, .size = paramsBuffer->getSize()};
+				infos[1].desc = smart_refctd_ptr(tempDataBuffer);
+				infos[1].info.buffer = {.offset = 0, .size = tempDataBuffer->getSize()};
+				infos[2].desc = smart_refctd_ptr(dataBuffer);
+				infos[2].info.buffer = {.offset = 0, .size = dataBuffer->getSize()};			
+				infos[3].desc = smart_refctd_ptr(globalHistogramsBuffer);
+				infos[3].info.buffer = {.offset = 0, .size = globalHistogramsBuffer->getSize()};
+				infos[4].desc = smart_refctd_ptr(partitionHistogramBuffer);
+				infos[4].info.buffer = {.offset = 0, .size = partitionHistogramBuffer->getSize()};
+				IGPUDescriptorSet::SWriteDescriptorSet writes[5] = {
+					{.dstSet = m_radixSortDs[1].get(), .binding = 0, .arrayElement = 0, .count = 1, .info = &infos[0]},
+					{.dstSet = m_radixSortDs[1].get(), .binding = 1, .arrayElement = 0, .count = 1, .info = &infos[1]},
+					{.dstSet = m_radixSortDs[1].get(), .binding = 2, .arrayElement = 0, .count = 1, .info = &infos[2]},
+					{.dstSet = m_radixSortDs[1].get(), .binding = 3, .arrayElement = 0, .count = 1, .info = &infos[3]},
+					{.dstSet = m_radixSortDs[1].get(), .binding = 4, .arrayElement = 0, .count = 1, .info = &infos[4]}
+				};
+				m_device->updateDescriptorSets(std::span(writes, 5), {});
+			}
```

Comment:
one does not simply overwrite the only copy of a descriptor set on every commandbuffer record


Think, what happens if you want 2 frames in flight, and a previous command buffer that's using the sets is still PENDING while you're recording this one?

Because you only have one copy of the descriptor set  you end up overwriting descriptors in the set which can potentially be currently used!